serve/graph/schema.resolvers.go

```diff
@@ -0,0 +1,150 @@
+package graph
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+// Code generated by github.com/99designs/gqlgen version v0.17.43
+
+import (
+	"context"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/gnolang/tx-indexer/serve/graph/model"
+	"github.com/gnolang/tx-indexer/types"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+)
+
+// Transactions is the resolver for the transactions field.
+func (r *queryResolver) Transactions(ctx context.Context, filter model.TransactionFilter) ([]*model.Transaction, error) {
+	it, err := r.
+		store.
+		TxIterator(
+			int64(dereferenceInt(filter.FromBlockHeight)),
+			int64(dereferenceInt(filter.ToBlockHeight)),
+			uint32(dereferenceInt(filter.FromIndex)),
+			uint32(dereferenceInt(filter.ToIndex)),
+		)
+	if err != nil {
+		return nil, gqlerror.Wrap(err)
+	}
+	defer it.Close()
+
+	fgw := dereferenceInt(filter.FromGasUsed)
+	tgw := dereferenceInt(filter.ToGasWanted)
+	fgu := dereferenceInt(filter.FromGasUsed)
+	tgu := dereferenceInt(filter.ToGasUsed)
+
+	var out []*model.Transaction
+	i := 0
+	for {
+		if i == maxElementsPerQuery {
```

Comment:
maxElementsPerQuery is always `!= 0`, so I suggest placing this check after `i++`

Reply:
I do not understand. I prefer to have error and special case checks at the top of the for loop, making clear what is the code branch doing all the real work.